raft/src/main/java/org/apache/kafka/raft/FollowerState.java

```diff
@@ -137,11 +137,11 @@ public boolean updateHighWatermark(OptionalLong highWatermark) {
         return fetchingSnapshot;
     }
 
-    public void setFetchingSnapshot(Optional<RawSnapshotWriter> fetchingSnapshot) {
+    public void setFetchingSnapshot(Optional<RawSnapshotWriter> newSnapshot) {
```

Comment:
Verified that a few of the tests failed without this and the changes to `MockRawSnapshotWriter` in this PR.